harry-core/src/harry/dsl/HistoryBuilder.java

```diff
@@ -19,598 +19,511 @@
 package harry.dsl;
 
 import java.util.*;
-import java.util.function.Consumer;
 import java.util.function.LongSupplier;
 
-import harry.core.Run;
+import harry.core.Configuration;
+import harry.core.MetricReporter;
+import harry.ddl.SchemaSpec;
+import harry.generators.EntropySource;
+import harry.generators.JdkRandomEntropySource;
+import harry.model.Model;
 import harry.model.OpSelectors;
+import harry.model.QuiescentChecker;
+import harry.model.clock.ApproximateMonotonicClock;
+import harry.model.sut.SystemUnderTest;
+import harry.operations.Query;
+import harry.reconciler.Reconciler;
+import harry.runner.DataTracker;
 import harry.visitors.MutatingRowVisitor;
 import harry.visitors.MutatingVisitor;
 import harry.visitors.ReplayingVisitor;
 import harry.visitors.VisitExecutor;
 
-import static harry.model.OpSelectors.DefaultPdSelector.PARTITION_DESCRIPTOR_STREAM_ID;
+// TODO: we can implement a pluggable time source via a custom clock, too; the only limitation is that order
+// of timestamps has to be consistent with LTS order
```

Comment:
Don't all Harry clocks have this requirement? OffsetClock and ApproximateMonotonicClock both provide monotonicity of RTS given monotonic LTSs.

Reply:
Yes, sure. The idea here is that you can come with your own timestamps, not have Harry generate them. Maybe for some weird edge condition testing or something.